packages/embark-mocha-tests/src/lib/index.js

```diff
@@ -83,82 +106,79 @@ class MochaTestRunner {
+    async.waterfall([
+      (next) => {
+        this.plugins.emitAndRunActionsForEvent('tests:contracts:compile:before', contractFiles, next);
       },
-      (cf, next) => { // compile contracts
-        events.request("compiler:contracts:compile", cf, next);
+      (_contractFiles, next) => {
+        contractFiles = _contractFiles;
+        events.request("compiler:contracts:compile", _contractFiles, next);
       },
-      (cc, next) => {
-        plugins.emitAndRunActionsForEvent('tests:contracts:compile:after', cc, next);
+      (_compiledContracts, next) => {
+        this.plugins.emitAndRunActionsForEvent('tests:contracts:compile:after', _compiledContracts, next);
       },
-      (cc, next) => { // override require
-        compiledContracts = cc;
-
-        Module.prototype.require = function(req) {
-          const prefix = "Embark/contracts/";
-          if (!req.startsWith(prefix)) {
-            return originalRequire.apply(this, arguments);
-          }
-
-          const contractClass = req.replace(prefix, "");
-          const instance = compiledContracts[contractClass];
-
-          if (!instance) {
-            throw new Error(`Cannot find module '${req}'`);
-          }
-
-          return instance;
-        };
-        next();
-      },
-      (next) => { // initialize Mocha
-        const mocha = new Mocha();
-
-        mocha.reporter(Reporter, { reporter: reporter });
-        const describeWithAccounts = (scenario, cb) => {
-          Mocha.describe(scenario, cb.bind(mocha, accounts));
-        };
-
-        mocha.suite.on('pre-require', () => {
-          global.describe = describeWithAccounts;
-          global.contract = describeWithAccounts;
-          global.assert = assert;
-          global.config = config;
+      (_compiledContracts, next) => {
+      compiledContracts = _compiledContracts;
+        const fns = this.files.map((file) => {
+          return (seriesCb) => {
+            fs.readFile(file, (err, data) => {
+              if (err) {
+                self.logger.error(__('Error reading file %s', file));
+                self.logger.error(err);
+                seriesCb(null, 1);
+              }
+              if (data.toString().search(/contract\(|describe\(/) === -1) {
+                return seriesCb(null, 0);
+              }
+
+              Module.prototype.require = function(req) {
+                const prefix = "Embark/contracts/";
+                if (!req.startsWith(prefix)) {
+                  return originalRequire.apply(this, arguments);
+                }
+
+                const contractClass = req.replace(prefix, "");
+                const instance = compiledContracts[contractClass];
+
+                if (!instance) {
+                  throw new Error(`Cannot find module '${req}'`);
+                }
+                return instance;
+              };
+
+              const mocha = new Mocha();
```

Comment:
We just want to run one instance of mocha. This reverts all the optimizations that have been done on the mocha tests.